ADR/pipeline-service/adr-0001.md

```diff
@@ -0,0 +1,46 @@
+# Service Account configuration for image builds in AppStudio 
+
+Date: Oct 5, 2022
+
+## Status
+
+Proposed
+
+## Context
+
+The `serviceAccount` used for running a `PipelineRun` workload in a workload cluster needs to have the right SCCs and linked credentials to be able to successfully execute a container image build. The linked credentials will mostly originate from the KCP workspace where the `PipelineRun` was synced from.
+Additionally, the SCC association needed for elevating capabilities required for image builds isn't something we'd prefer to expose the user to.
+
+
+
+## Decision
+
+* As a Pipeline Service platform feature, any service/user intending to build a container image would need to create/use the "image-builder" service account in the KCP workspace. 
+Use of any other service accounts will not lead to a successful assembling of the image.
+
+* The AppStudio Build controller would tap into this platform contract to create relevant PipelineRuns for building container images.
+
+## Consequences
+
+
+### Implementation - AppStudio Build Service Controller
+
+* The AppStudio Build controller creates an "image-builder" service account in the KCP workspace/namespace.
```

Comment:
can jvm-build-service reuse the SA created by appstudio build controller, or does it have to separately do similar actions @sbose78 ?

@stuartwdouglas FYI

Reply:
If the service is always used in scoe of AppStudio or HACBS, I don't see a reason why you would need to create another service account.
If the service was to become standalone, then I understand why you would need to manage your own service account.

Reply:
yep my assumption/hope was that "share/reuse" would be the answer, but I did not want to assume / get explicit confirmation / consensus

Reply:
jvm-build-service or any other service except build-service should not edit/use linked secrets of the service account.

Reply:
So we need a decision @sbose78 on what services outside of appstudio build service should be doing here

Reply:
Let me ponder on this a little and respond.